services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankeMapper.kt

```diff
@@ -0,0 +1,64 @@
+package fi.hel.haitaton.hanke
+
+import fi.hel.haitaton.hanke.ContactType.MUU
+import fi.hel.haitaton.hanke.ContactType.OMISTAJA
+import fi.hel.haitaton.hanke.ContactType.RAKENNUTTAJA
+import fi.hel.haitaton.hanke.ContactType.TOTEUTTAJA
+import fi.hel.haitaton.hanke.domain.Hanke
+import fi.hel.haitaton.hanke.domain.HankeYhteystieto
+import fi.hel.haitaton.hanke.domain.Hankealue
+import fi.hel.haitaton.hanke.geometria.GeometriatService
+import org.springframework.stereotype.Component
+
+@Component
+class HankeMapper(private val geometriatService: GeometriatService) {
+
+    fun domainFromEntity(entity: HankeEntity): Hanke =
```

Comment:
I was kind of hoping that this would replace the domain object creation in the service (but not in this pr). The service implementation just seems messy. But yes, it is even worse to have two different domain object creation styles. I can copypaste the service implementation to the mapper.

Reply:
Changed hanke service to use this new mapper.